frontend/lambda-in/src/Components/DevInfoEditz/DevInfoEditz.js

```diff
@@ -1,38 +1,221 @@
 import Button from '@material-ui/core/Button';
 import Paper from '@material-ui/core/Paper';
 import Typography from '@material-ui/core/Typography';
-import TextField from '@material-ui/core/TextField';
-import ExpansionPanel from '@material-ui/core/ExpansionPanel';
-import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
-import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
-import Icon from '@material-ui/core/Icon';
 import axios from 'axios';
 import './DevInfoEditz.css';
+import BasicInfo from '../utilityComponents/SeekerEditUtils/BasicInfo';
+import SocialLinks from '../utilityComponents/SeekerEditUtils/SocialLinks';
 import BioSkills from '../utilityComponents/SeekerEditUtils/BioSkills';
 import Projects from '../utilityComponents/SeekerEditUtils/Projects';
-import Education from '../utilityComponents/SeekerEditUtils/Education';
 import Experience from '../utilityComponents/SeekerEditUtils/Experience';
-import SocialLinks from '../utilityComponents/SeekerEditUtils/SocialLinks';
-import BasicInfo from '../utilityComponents/SeekerEditUtils/BasicInfo';
+import Education from '../utilityComponents/SeekerEditUtils/Education';
+
+/**
+ * Form handling user profile updates
+ *
+ * @description This component handle in a local-state all data modification.
+ * The global state is updeated only when the PUT request response with status 200
+ */
+class DevInfoEdit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ready: false,
+    };
+    this.handleOnDeleteItem = this.handleOnDeleteItem.bind(this);
+    this.handleCreateItem = this.handleCreateItem.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.setFormState = this.setFormState.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+    this.update = this.update.bind(this);
+  }
+
+  /**
+   * Get a copy of user's data to keep global state inmutable.
+   */
+  componentDidMount() {
+    // eslint-disable-next-line react/prop-types
+    const { getGS } = this.props;
+    // // console.log('FORM', this.props);
+    this.userStateCopy = { ...getGS('userInfo') };
+    this.setState({ ready: true, ...getGS('userInfo') });
+
+    /**
+     * Listen for MapChip's custom event emmited when a item is deleted.
+     */
+    // eslint-disable-next-line react/no-find-dom-node
+    ReactDOM.findDOMNode(this).addEventListener(
+      'onDeleteItem',
+      this.handleOnDeleteItem,
+    );
+
+    // eslint-disable-next-line react/no-find-dom-node
+    ReactDOM.findDOMNode(this).addEventListener(
+      'onCreateItem',
+      this.handleCreateItem,
+    );
+  }
+
+  componentWillUnmount() {
+    // eslint-disable-next-line react/no-find-dom-node
+    ReactDOM.findDOMNode(this).removeEventListener('onDeleteItem', () => null);
+    // eslint-disable-next-line react/no-find-dom-node
+    ReactDOM.findDOMNode(this).removeEventListener('onCreateItem', () => null);
+  }
+
```

Comment:
I am concerned that this does not fall within best practices for React. Specifically when referring to https://reactjs.org/docs/react-dom.html#finddomnode and https://reactjs.org/docs/handling-events.html.

> In most cases, you can attach a ref to the DOM node and avoid using findDOMNode at all.

> findDOMNode is an escape hatch used to access the underlying DOM node. In most cases, use of this escape hatch is discouraged because it pierces the component abstraction

> When using React you should generally not need to call addEventListener to add listeners to a DOM element after it is created. Instead, just provide a listener when the element is initially rendered.

Why were you not able to pass down event handlers and attach them on render?